src/components/websocket/Websocket.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import { UPDATE_TYPE_HEADER } from 'components/study-container';
+import { useCallback, useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import ReconnectingWebSocket from 'reconnecting-websocket';
+import { closeWebsocket, openWebsocket } from 'redux/actionsTyped';
+import { ReduxState } from 'redux/reducer.type';
+import { getUrlWithToken, getWsBase } from 'services/utils';
+
+// the delay before we consider the WS truly connected
+const DELAY_BEFORE_WEBSOCKET_CONNECTED = 12000;
+
+const PREFIX_STUDY_NOTIFICATION_WS =
+    import.meta.env.VITE_WS_GATEWAY + '/study-notification';
+
+const extractUpdateTypeFromHearder = (eventData: {
+    [k: string]: unknown;
+}): string | undefined => {
+    if (
+        'headers' in eventData &&
+        eventData.headers &&
+        typeof eventData.headers === 'object'
+    ) {
+        if (
+            UPDATE_TYPE_HEADER in eventData.headers &&
+            typeof eventData.headers.updateType === 'string'
```

Comment:
```suggestion
            typeof eventData.headers?.updateType === 'string'
```
we can now even combine this if condition with the one above